### DISCUSSION TITLE
Python 3.11

### DISCUSSION THREAD

**Discussion by RiseT** [stefan-jansen/zipline-reloaded#176] [Q&A]
Just wondering: Zipline 2.4 doesn't install with Python 3.11 right now, although wheels for 3.11 is available. Is this expected or just me?

`mamba search zipline-reloaded`

```
Loading channels: done
# Name                       Version           Build  Channel
zipline-reloaded                 2.4 py310h9b08ddd_0  conda-forge
zipline-reloaded                 2.4 py310h9b08ddd_1  conda-forge
zipline-reloaded                 2.4  py38hbaf524b_0  conda-forge
zipline-reloaded                 2.4  py38hbaf524b_1  conda-forge
zipline-reloaded                 2.4  py39hc266a54_0  conda-forge
zipline-reloaded                 2.4  py39hc266a54_1  conda-forge
```

`mamba update python=3.11`

```
Looking for: ['python=3.11']

conda-forge/win-64                                          Using cache
conda-forge/noarch                                          Using cache
Could not solve for environment specs
Encountered problems while solving:
  - package zipline-reloaded-2.4-py310h9b08ddd_1 requires python_abi 3.10.* *_cp310, but none of the providers can be installed

The environment can't be solved, aborting the operation
```

`mamba list python`

```
# packages in environment at C:\Users\Rise\mambaforge\envs\zl:
#
# Name                    Version                   Build  Channel
ipython                   8.11.0             pyh08f2357_0    conda-forge
python                    3.10.9          h4de0772_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-interface          1.6.0                      py_0    conda-forge
python_abi                3.10                    3_cp310    conda-forge
```

**Comment by riverallzero** [comment]
Hi @RiseT ,
The Zipline package currently does not support Python 3.11, and this is likely the reason you are experiencing issues installing it. While there may be wheels available for Python 3.11, they may not be compatible with the package due to changes made in the new Python version.

**Comment by RiseT** [comment]
Hey Dayoung,

Thank you very much for your reply! I unfortunately don't know much (if anything) about Wheels, but was refering to the following two pieces of information, trying to add 1 and 1 together:

https://github.com/stefan-jansen/zipline-reloaded/pull/159

https://github.com/h5py/h5py/issues/2146#issuecomment-1400125971

**Comment by riverallzero** [comment]
I think if you upgrade h5py package to latest version, Zipline 2.4 will install on python 3.11.

Do `pip install --upgrade h5py`, then reinstall Zipline 2.4!!

**Comment by RiseT** [comment]
Hi Dayoung,

While I'm using Conda (or rather its replacement Mamba) as the package manager instead of pip, the version of h5py installed in this environment is the latest already:

```
# packages in environment at C:\Users\Rise\mambaforge\envs\zl:
#
# Name                    Version                   Build  Channel
h5py                      3.8.0           nompi_py310h727aef9_101    conda-forge
```

**Comment by riverallzero** [comment]
Then i think you should downgrade version 3.11 to 3.10 or less.
I recommend you to use 3.7 version. It is relatively stable among Python 3 versions and many libraries and frameworks support 3.7 versions. If you use 3.7 version there won't be happen install issues.

**Comment by RiseT** [comment]
Hey Dayoung,

Thanks for caring and your efforts! I had (and still have) zipline-reloaded 2.4 running with Python 3.10 successfully, but since it looked like (see the two links above) it should run with 3.11, I wanted to let the devs know that it doesn't (at least for me, that is).

I guess I've found the specific reasons today when [reporting a (mamba) bug](https://github.com/mamba-org/mamba/issues/2414#issuecomment-1490080102) (which in fact turned out to be an actual bug) to the mamba devs, though. Here's the corresponding excerpt of one of the dev's reply:

> It looks like zipline-reloaded is not made noarch: python which means it has to be built for every version of Python, which has not happened yet for Python 3.11.
> **From a quick look, it seems unnecessary.** You could ask the maintainer to make the package noarch: python there:
> https://github.com/conda-forge/zipline-reloaded-feedstock

cc: @stefan-jansen
